### PR TITLE
Build MAUI for MacCatalyst

### PR DESCRIPTION
**Pull request by Redth**

### CODE REVIEW

File: src/Compatibility/Core/src/iOS/Renderers/OpenGLViewRenderer.cs [hunk 1]
```diff
@@ -90,7 +95,7 @@ void SetupRenderLoop(bool oneShot)
 			_displayLink.AddToRunLoop(NSRunLoop.Current, NSRunLoop.NSRunLoopCommonModes);
 		}
 
-		class Delegate : GLKViewDelegate
+		class Delegate : GLKViewDelegate, IGLKViewDelegate
```
Comment (mattleibow) [inline src/Compatibility/Core/src/iOS/Renderers/OpenGLViewRenderer.cs:98]: Why is this needed? Doesn't `GLKViewDelegate` already implement `IGLKViewDelegate`?

File: src/Compatibility/Core/src/iOS/CollectionView/CarouselViewController.cs [hunk 2]
```diff
@@ -101,13 +101,23 @@ void BoundsSizeChanged()
 			Carousel.ScrollTo(Carousel.Position, position: Microsoft.Maui.Controls.ScrollToPosition.Center, animate: false);
 		}
 
+#if MACCATALYST
+		[Foundation.Export("scrollViewWillBeginDragging:")]
+		public virtual void DraggingStarted(UIScrollView scrollView)
+#else
 		public override void DraggingStarted(UIScrollView scrollView)
+#endif
```
Comment (mattleibow) [inline src/Compatibility/Core/src/iOS/CollectionView/CarouselViewController.cs:109]: I need to file a bug on this.

File: src/Compatibility/Core/src/iOS/Renderers/TabbedRenderer.cs [hunk 3]
```diff
@@ -402,12 +402,19 @@ void UpdateBarTextColor()
...
-				attributes.TextColor = barTextColor.ToUIColor();
+				tabBarTextColor = barTextColor.ToUIColor();
+
+#if MACCATALYST
+			var attributes = new UIStringAttributes();
+			attributes.ForegroundColor = tabBarTextColor;
+#else
+			var attributes = new UITextAttributes();
+			attributes.TextColor = tabBarTextColor;
+#endif
```
Comment (mattleibow) [inline src/Compatibility/Core/src/iOS/Renderers/TabbedRenderer.cs:417]: We can prob make this always use `UIStringAttributes` since UITextAttributes was added in iOS 6

File: src/Controls/samples/Controls.Sample.MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
+{
+  "images": [
```
Comment (mattleibow) [inline src/Controls/samples/Controls.Sample.MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json:2]: Should this be resizetized?

File: src/Controls/samples/Controls.Sample.SingleProject/MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
+{
+  "images": [
```
Comment (mattleibow) [inline src/Controls/samples/Controls.Sample.SingleProject/MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json:2]: And this be resizetized?

File: src/Compatibility/Core/src/iOS/Renderers/OpenGLViewRenderer.cs (hunk 1 above)
Comment (Redth) [reply src/Compatibility/Core/src/iOS/Renderers/OpenGLViewRenderer.cs:98]: It was complaining about not being able to implicitly cast and this made the compiler happy

File: src/Controls/samples/Controls.Sample.SingleProject/MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json (hunk 5 above)
Comment (Redth) [reply src/Controls/samples/Controls.Sample.SingleProject/MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json:2]: Eventually yes

File: src/Controls/samples/Controls.Sample.MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json (hunk 4 above)
Comment (Redth) [reply src/Controls/samples/Controls.Sample.MacCatalyst/Assets.xcassets/AppIcon.appiconset/Contents.json:2]: Eventually yes
